src/io/mod.rs

```diff
@@ -74,6 +77,33 @@ mod imp {
         ($fmt:expr, $($arg:tt)*) => (print!(concat!($fmt, "\n"), $($arg)*));
     }
 
+    struct DMASerial {
```

Comment:
Moving all these implementations into it's own module seems like it would make the file a bit cleaner.